### PR TITLE
Improved Transaction Handling and Agreement Creation on EVM and Solana

### PR DESCRIPTION
**Pull request by nomadbitcoin**
This Pull Request encapsulates a range of enhancements, bug fixes, and new feature implementations aimed at improving the overall functionality and user experience of our platform. The primary focus is on the `AddAgreement` functionality, now extended to support both local EVM and Solana networks. Below is a breakdown of the contributions:

   - **Solana Packages Setup**: Installed essential Solana packages and updated the README with setup instructions to facilitate a smoother onboarding process.
   - **Bug Fixes**: Addressed import issues and other minor bugs to adapt to the new folder structure, ensuring a stable and consistent development environment.
   - **Deployment Scripts for Solana**: Developed scripts for deployments and agreement creations.
   - **Transaction Handling Improvements**: Enhanced transaction handling mechanisms, including creating hooks for transaction management and refactoring contract initialization and invocation processes.
   - **Frontend Enhancements**: Worked on improving frontend features such as better toast messages and chain verification for Ethereum to provide a more intuitive user interface.

By @Jaxiii:
   - Created the Solana agreement contract.
   - **Solana Tests**: Created scripts for testing and deployment on Solana Devnet.
   - **Solana Payment Features**: Implemented and tested fake token creation and payment processing on Solana.

By @mikaelcarrara:
   - **Frontend Style Improvements**: Enhanced flash messages, color schemes, and translated frontend texts to provide a more user-friendly interface.
   - **Mobile UI Adjustments**: Made necessary adjustments to the mobile UI including the menu and footer for a better mobile user experience.
   - **Frontend View Enhancements**: Worked on improving the views related to payment history and dashboard, making information more accessible and easier to understand.

The significant upgrade in the `AddAgreement` functionality paves the way for robust transaction handling across multiple networks. While this update mainly addresses the transaction handling in `AddAgreement`, other interactions within the platform remain outdated and will be tackled in the upcoming PRs.

This PR closes issues #41, #73, #74, and #56.

### CODE REVIEW

File: chains/solana/transactions/addAgreement.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    [stringBuffer, companyAccount.toBuffer()],
+    details.contract.programId
+  );
+
+  try {
+    const agreementAddress = anchor.web3.Keypair.generate();
+
+    const tx = await details.contract.methods
+      .initializeAgreement(agreementData).accounts({
+          agreement: agreementAddress.publicKey,
```
Comment (danicuki) [inline chains/solana/transactions/addAgreement.js:26]: there is an important info missing here: which community this professional / agreement belongs

File: chains/ethereum/transactions/addAgreement.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export const addAgreement = async (details) => {
+  try {
+    const paymentAmountInWei = ethers.utils.parseUnits(details.paymentAmount.toString(), 18);
+
+    const tx = await details.contract.createAgreement(
+      details.title,
+      details.description,
+      details.professional,
+      details.skills,
```
Comment (danicuki) [inline chains/ethereum/transactions/addAgreement.js:11]: we missed the information about which community this agreement belongs to. We should design this, also in the contract (have a community registry, etc)

File: components/AddAgreement/AddAgreement.js [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
 import styles from "./AddAgreement.module.scss"
 import Image from 'next/image'
-import { BeatLoader } from "react-spinners"
-import { ethers } from "ethers"
+import Loader from '../utils/Loader';
+import Toast from '../utils/Toast';
+import useTransactionHandler from '../../hooks/useTransactionHandler';
 import { useTranslation } from "react-i18next"
 import * as Yup from "yup"
+import { useWallet } from '@solana/wallet-adapter-react';
```
Comment (danicuki) [inline components/AddAgreement/AddAgreement.js:11]: is there a way to make this component chain agnostic?

File: components/utils/Toast/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (transactionPending) {
+    return (
+      <div className="flash-pending transaction-info">
+        <p>
+          <Image src="/pending-icon.svg" width={20} height={20} alt="Pending icon" />
+          {t("agreement-pending")}
+        </p>
+          <a href={`https://polygonscan.com/tx/${transactionHash}`} target="_blank" rel="noopener noreferrer">
+            Check the status here
```
Comment (danicuki) [inline components/utils/Toast/index.js:27]: missing translation

File: contracts/solana/Anchor.toml [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
-agreement_program = "Fz3wjVypwyrY4a3fad9gTpDotqZg2bFBwEW4TkyruMTo"
+agreement_program = "FVz7RJ6H6zFUkTx1sGuCDHDmYr96EQHR4g52xTmL8ZPn"
 
 [registry]
 url = "https://api.apr.dev"
 
 [provider]
 cluster = "Localnet"
-wallet = "/Users/nomadbitcoin/.config/solana/id.json"
+wallet = "/Users/brunobarros/.config/solana/id.json"
```
Comment (danicuki) [inline contracts/solana/Anchor.toml:12]: this should be dev agnostic

File: contracts/solana/README.md [hunk 6]
```diff
@@ -31,4 +31,56 @@ solana-test-validator
...
\ No newline at end of file
+If you are using a Mac, be aware that you may need to install some additional dependencies, such as `gnu-tar`.
+
+## Setting Up Wallet
+
+To create agreements, you'll need to set up your wallet. Export your private key from Phantom and save it as `PHANTOM_WALLET_PRIVATE_KEY` in the root `.env.development.local` file. Then, run the convert script to transform it into a key pair that is required by most Anchor scripts.
+
+```sh
+# Run the conversion script (replace with the correct command if necessary)
+./convert-script.sh
```
Comment (danicuki) [inline contracts/solana/README.md:42]: where is this script in the code? I did not find it

File: package.json [hunk 7]
```diff
@@ -9,6 +9,8 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@coral-xyz/anchor": "^0.28.1-beta.2",
```
Comment (danicuki) [inline package.json:12]: why two different versions of anchor?

File: public/allowedTokens.json [hunk 8]
```diff
@@ -18,7 +18,7 @@
     "decimals": 6
   },
   {
-    "address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+    "address": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
```
Comment (danicuki) [inline public/allowedTokens.json:21]: again and again and again this f*ng file

File: public/locales/en-US/translation.json [hunk 9]
```diff
@@ -1,9 +1,36 @@
 {
   "dashboard": "Dashboard",
   "agreements": "Agreements",
+
```
Comment (danicuki) [inline public/locales/en-US/translation.json:4]: this file is starting to get a mess? Let's try to make it more organized by contexts? 
global terms could be on the root level, but screen specific could follow some rule.
Also, make keys alphabetically ordered, would be much better

**Review by danicuki — APPROVED** [review]

File: chains/ethereum/transactions/addAgreement.js (hunk 2 above)
Comment (Jaxiii) [reply chains/ethereum/transactions/addAgreement.js:11]: Noted, will add this feature to the Solana program.

File: chains/solana/transactions/addAgreement.js (hunk 1 above)
Comment (Jaxiii) [reply chains/solana/transactions/addAgreement.js:26]: Noted, will add this feature to the Solana program.

**Review by Jaxiii — COMMENTED** [review]
@danicuki comments were noted to future updates on the Solana program.

**Pull request merged**